test/Microsoft.AspNetCore.Mvc.DataAnnotations.Test/Internal/DataAnnotationsMetadataProviderTest.cs

```diff
@@ -180,14 +194,21 @@ public void CreateDisplayMetadata_DisplayAttribute_NameFromResources()
 
             // Assert
             Assert.Equal("name from resources", context.DisplayMetadata.DisplayName());
+
+            if (stringLocalizerFactory != null)
```

Comment:
I agree. The added verification is unnecessary: If the configured `DisplayName()` lambda is returning the correct value, you're done.